tests/e2e/operator/operator_test.go

```diff
@@ -384,3 +398,22 @@ func LogDebugInfo() {
 	}
 	GinkgoWriter.Println("Logs from istiod pod: \n", logs)
 }
+
+// getKey returns the client.ObjectKey for the given object
+func getKey(namespace, name string) client.ObjectKey {
+	return client.ObjectKey{Namespace: namespace, Name: name}
+}
+
+// getCRDKey returns the client.ObjectKey for the given CRD name
+func getCRDKey(name string) client.ObjectKey {
```

Comment:
We need a better name for this, since this isn't only for CRDs. For example, you have the following call in one of the tests: `getCRDKey(controlPlaneNamespace)`, which is confusing. A better name might be `getClusterKey`.

But ideally we only want to have one function for both namespaced and cluster-scoped resources. Maybe we can (ab)use varargs for this:
```
func key(name string, namespace ...string) client.ObjectKey {
	if len(namespace) > 1 {
		panic("you can only provide one namespace")
	} else if len(namespace) == 1 {
		return client.ObjectKey{Name: name, Namespace: namespace[0]}
	} 
	return client.ObjectKey{Name: name}
}
```

Although I don't like abusing language features like this, maybe it's not that bad in this context. Especially if we explain why in the function comment.

Also, I think we can call this function just `key` instead of `getKey`. It should make the tests more readable, and it's pretty clear what `key(podName, podNamespace)` does.